conans/client/conf/__init__.py

```diff
@@ -97,5 +97,36 @@ def proxies(self):
     @property
     def settings_defaults(self):
         default_settings = self.get_conf("settings_defaults")
+        default_settings = self._mix_settings_with_env(default_settings)
         values = Values.from_list(default_settings)
         return values
+
+    def _mix_settings_with_env(self, settings):
+        """Reads CONAN_ENV_XXXX variables from environment
+        and if it's defined uses these value instead of the default
+        from conf file. If you specify a compiler with ENV variable you
+        need to specify all the subsettings, the file defaulted will be
+        ignored"""
+
+        def get_env_value(name):
+            env_name = "CONAN_ENV_%s" % name.upper().replace(".", "_")
+            return os.getenv(env_name, None)
+
+        def get_setting_name(env_name):
+            return env_name[10:].lower().replace("_", ".")
+
+        ret = []
```

Comment:
I'm not sure, with the list I'm keeping the original settings order, the dict will be unpredictable non-ordered.